File: backend/app/domain/entities/notification.py
from dataclasses import dataclass


@dataclass(frozen=True)
class Notification:
    """Alerta calculada al vuelo para el panel de notificaciones (campana
    del header) — no se persiste en ninguna tabla, se recalcula cada vez
    que se abre el desplegable. `message` ya viene armado en texto listo
    para mostrar; el frontend solo elige ícono/color según `type`."""

    type: str  # 'preceptor_temp_assignment' | 'schedule_exception' | 'consecutive_absences'
    message: str

File: frontend/lib/domain/entities/notification.dart
/// Domain entity: AppNotification
///
/// Alerta calculada al vuelo por el backend para el panel de
/// notificaciones (campana del header) — no se persiste ni tiene estado
/// de leído/no leído, se recalcula cada vez que se abre el desplegable.
class AppNotification {
  const AppNotification({required this.type, required this.message});

  final String type; // 'preceptor_temp_assignment' | 'schedule_exception' | 'consecutive_absences'
  final String message;

  factory AppNotification.fromJson(Map<String, dynamic> json) => AppNotification(
        type: json['type'],
        message: json['message'],
      );
}
